### DISCUSSION TITLE
Single cell with text in a column of numbers/doubles

### DISCUSSION THREAD

**Discussion by braunerphilipp** [davidgohel/flextable#499] [Ideas]
thanks for this great package! it's amazingly easy to use.

I have one problem/question/idea though. I have a column of doubles and in that a few strings. The double formatter refuses to work as it encounters the strings. What I want to have: It should print the correctly formatted doubles, if it's a string it should print out the string. As there any way to do that? Can I override the cell content after the formatting has been done?

minimal example: if valueA contains "text", the column is not formatted. If "text" is replaced with NA it works like a charm.

```
exampleTable <- tibble(
  labels = c("variable A", "variable B", "variable C", "variable D"),
  valueA = c(32.12345, "text", 4.12345, 3.412345),
  valueB = c(3.12345, 4.3, .25, 0.3212345))
exampleTable

flextable(exampleTable) %>%
  flextable::colformat_double(j=2:3, digits=3, nan_str="nan", na_str="na") %>%
  flextable::theme_box()
```

**Comment by davidgohel** [comment]
this happens because a vector of atomic values can't store different types of data (opposed to the list). This is out of flextable scope and is about R. 


If you want to format 
- characters, use `colformat_char()`
- double, use `colformat_double()`
- ...

See  functions like `colformat_char(), colformat_datetime(), colformat_double(), colformat_image(), colformat_int(), colformat_lgl(), colformat_num(), set_formatter()`


Function `compose()` (or `mk_par()` to avoid name conflict with purrr) let you format any column the way you want. That's probably what you need

**Comment by braunerphilipp** [comment]
Thanks! My workaround is to convert the column to a string using the right
formatter and replace the cells manually before passing it to flextable. It
works but is not as convenient as I thought it could be. But R is not Excel
and that's usually a good thing :)
